core/jvm/src/main/scala/zio/RuntimePlatformSpecific.scala

```diff
@@ -48,10 +48,24 @@ private[zio] trait RuntimePlatformSpecific {
 
   def enableLoomBasedExecutor(implicit trace: Trace): ZLayer[Any, LoomNotAvailableException, Unit] =
     ZLayer.suspend {
-      LoomSupport.newVirtualThreadPerTaskExecutor() match {
-        case None           => ZLayer.fail(LoomNotAvailableException("Loom API not available", null))
-        case Some(executor) => Runtime.setExecutor(Executor.fromJavaExecutor(executor))
-      }
+      createLoomExecutor.fold(
+        ZLayer.fail(_),
+        Runtime.setExecutor
+      )
+    }
+
+  def enableLoomBasedBlockingExecutor(implicit trace: Trace): ZLayer[Any, LoomNotAvailableException, Unit] =
+    ZLayer.suspend {
+      createLoomExecutor.fold(
+        ZLayer.fail(_),
+        Runtime.setBlockingExecutor
+      )
+    }
+
+  private def createLoomExecutor(implicit trace: Trace): Either[LoomNotAvailableException, Executor] =
```

Comment:
I think I would change this to a `lazy val` instead. This way if users use Loom for both executors, the same `Executor` instance will be reused which will reduce context switching as we'll simply continue execution on the current thread when `blocking` is used